src/domain/event/__tests__/CreateEventPage.test.tsx

```diff
@@ -481,6 +465,130 @@ const fillForm = async (eventFormData: Partial<CreateEventFormFields>) => {
+    ).not.toBeChecked();
+    expect(
+      within(languageSelector).getByLabelText(/englanti/i)
+    ).not.toBeChecked();
+  });
+
+  test('transletable fields appears in selected languages', async () => {
+    render(<CreateEventPage />, { mocks });
+    const languageSelector = await screen.findByTestId(
+      formLanguageSelectorTestId
+    );
+    // Select Swedish (with Finnish that is already selected)
+    userEvent.click(within(languageSelector).getByLabelText(/ruotsi/i));
+
+    transletableFieldLabels.forEach((labelText) => {
+      screen.getByLabelText(new RegExp(labelText.source + / \(FI\)/.source));
+      screen.getByLabelText(new RegExp(labelText.source + / \(SV\)/.source));
+    });
+
+    // Unselect Finnish
+    userEvent.click(within(languageSelector).getByLabelText(/suomi/i));
+    transletableFieldLabels.forEach((labelText) => {
+      expect(
+        screen.queryByLabelText(new RegExp(labelText.source + / \(FI\)/.source))
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  test.todo(
+    'filled fields for unselected languages are cleared when submitting the form'
+  );
+  // test('filled fields for unselected languages are cleared when submitting the form', async () => {
+  //   const omitUnselectedLanguagesFromValuesSpy = jest.spyOn(
+  //     Utils,
+  //     'omitUnselectedLanguagesFromValues'
+  //   );
+  //   const genericSwedishValue = 'SV translation';
+  //   jest.spyOn(Router, 'useParams').mockReturnValue({
+  //     id: eventId,
+  //   });
+  //   render(<CreateEventPage />, {
+  //     mocks: editMocks,
+  //   });
+  //   await waitFor(() => {
+  //     expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
+  //   });
+  //   const languageSelector = await screen.findByTestId(
+  //     formLanguageSelectorTestId
+  //   );
+  //   // There should be a value in Finnish field
+  //   expect(screen.getByLabelText(/Tapahtuman nimi \(FI\)/i)).toHaveValue(
+  //     eventName
+  //   );
+  //   // Select Swedish (with Finnish that is already selected)
+  //   userEvent.click(within(languageSelector).getByLabelText(/ruotsi/i));
+
+  //   // Populate Swedish fields
+  //   transletableFieldLabels.forEach((labelText) => {
+  //     userEvent.type(
+  //       screen.getByLabelText(new RegExp(labelText.source + / \(SV\)/.source)),
+  //       genericSwedishValue
+  //     );
+  //   });
+
+  //   // Unselect Swedish which was a newly filled field
+  //   userEvent.click(within(languageSelector).getByLabelText(/ruotsi/i));
+
+  //   // Submit the form and the Swedish values should not be sent
+  //   userEvent.click(
+  //     screen.getByRole('button', {
+  //       name: 'Tallenna ja siirry tapahtuma-aikoihin',
+  //     })
+  //   );
+  //   console.log(
+  //     'return values: ',
+  //     omitUnselectedLanguagesFromValuesSpy.mock.results[0]
+  //   );
+  //   expect(omitUnselectedLanguagesFromValuesSpy).toHaveBeenCalled();
+  // });
```

Comment:
How could I test that omitUnselectedLanguagesFromValues is called and the submit removes the unselected values?

Reply:
You need to have mutation mock variables that only has Finnish language for example, that mock should match when you have unselected languages

Reply:
You need to have mutation mock variables that only has Finnish language for example, that mock should match when you have unselected languages

Reply:
It's now done so that it relies to mocked response. I'm not happy about that, because it's unclear what the test really tests. However, it really testes the thing! 👍 If the spy on omitUnselectedLanguagesFromValues -method would work, the test would be clearer and faster. Same thing in both, the create and the edit page.